Readme.md

```diff
@@ -1,24 +1,108 @@
+
+    ```csharp
+    protected void ASPxScheduler1_PopupMenuShowing(object sender, DevExpress.Web.ASPxScheduler.PopupMenuShowingEventArgs e) {
+        e.Menu.ClientSideEvents.PopUp = "OnClientPopupMenuShowing";
+    
+        if(e.Menu.MenuId == DevExpress.XtraScheduler.SchedulerMenuItemId.AppointmentMenu) {
+            DevExpress.Web.MenuItem newItemCopy = new DevExpress.Web.MenuItem();
+            newItemCopy.Name = "CopyAppointment";
+            newItemCopy.Text = "Copy";
+            newItemCopy.ItemStyle.Font.Bold = true;
+            e.Menu.Items.Add(newItemCopy);
+            e.Menu.JSProperties["cpMenuName"] = "AppointmentMenu";
+        }
+        if(e.Menu.MenuId == DevExpress.XtraScheduler.SchedulerMenuItemId.DefaultMenu) {
+            DevExpress.Web.MenuItem newItemPaste = new DevExpress.Web.MenuItem();
+            newItemPaste.Name = "PasteAppointment";
+            newItemPaste.Text = "Paste";
+            newItemPaste.ItemStyle.Font.Bold = true;
+            e.Menu.Items.Insert(0, newItemPaste);
+            e.Menu.JSProperties["cpMenuName"] = "DefaultMenu";
+        }
+    }
+    ```
 
-<p>To implement the copy / paste appointment operation via an appointment context menu, the following approach is used in this sample:<br />1. Two custom items ("Copy" and "Paste") are added into the "view" and "appointment" context menus. This is implemented in the <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebASPxSchedulerASPxScheduler_PopupMenuShowingtopic">ASPxScheduler.PopupMenuShowing</a> event handler.<br /><br />2. The availability of these items is managed in the client-side Popup event handler of the mentioned menus (the "Paste" item must be disabled if there is no copied appointment). Please refer to the javascript <strong>OnClientPopupMenuShowing</strong> method.<br /><br />3. When the "Copy" item is clicked, the ID of a currently selected appointment is stored as a HiddedField value. When the "Paste" item is clicked, a custom callback operation is executed, which results in copying the selected appointment on the server side.<br />
+2. The context menu's [PopUp](https://docs.devexpress.com/AspNet/DevExpress.Web.MenuClientSideEvents.PopUp) event handler disables **Paste** command when there is no copied appointment.
 
-<br/>
+    ```jscript
+    function OnClientPopupMenuShowing(s, e) {
+        for (menuItemId in e.item.items) {
+            if (e.item.items[menuItemId].name == "PasteAppointment") {
+                e.item.items[menuItemId].SetEnabled(document.getElementById("hdCopiedAppointmentID").value != '');
+            }
+        }
+    }
+    ```
 
+3. When a user clicks the **Copy** command, the ID of the currently selected appointment is saved to `HiddedField`.
```

Comment:
Is `HiddedField` a typo? The reason I ask is that in conversation normally we would say "hidden field".

Reply:
```suggestion
3. When a user clicks the **Copy** command, the ID of the currently selected appointment is saved to `HiddenField`.
```